src/clients/ProfitClient.ts

```
@@ -338,7 +361,20 @@ export class ProfitClient {
     const scaledInputAmount = deposit.inputAmount.mul(inputTokenScalar);
     const inputAmountUsd = scaledInputAmount.mul(inputTokenPriceUsd).div(fixedPoint);
 
-    const outputTokenInfo = hubPoolClient.getL1TokenInfoForL2Token(deposit.outputToken, deposit.destinationChainId);
+    // Unlike the input token, output token is not always resolvable via HubPoolClient since outputToken
+    // can be any arbitrary token.
+    let outputTokenInfo: L1Token;
+    // If the output token and the input token are equivalent, then we can look up the token info
+    // via the HubPoolClient since the output token is mapped via PoolRebalanceRoute to the HubPool.
+    // If not, then we should look up outputToken in the TOKEN_SYMBOLS_MAP for the destination chain.
+    const matchingTokens =
+      TOKEN_SYMBOLS_MAP[inputTokenInfo.symbol]?.addresses[deposit.destinationChainId] === deposit.outputToken;
+    if (matchingTokens) {
+      outputTokenInfo = hubPoolClient.getL1TokenInfoForL2Token(deposit.outputToken, deposit.destinationChainId);
+    } else {
+      // This function will throw if the token is not found in the TOKEN_SYMBOLS_MAP for the destination chain.
```

fwiw, if profitability computation throws then we just assume the fill is unprofitable, so it's not necessarily a problem to throw here :+1:

---

I do worry about silent errors though and we'll end up just seeing lots of "unprofitable deposits"

---

The ProfitClient propagates the underlying error through to the "fill is unprofitable" message, so it is possible to see it in even the slack logs.

I don't really see how we can handle it otherwise, since if it threw "hard" then any depositor could take the relayer down by making a deposit with some random dog coin as the outputToken.